Makefile

```diff
@@ -2,30 +2,42 @@ PREFIX            ?= $(shell pwd)
 FILES             ?= $(shell find . -type f -name '*.go' -not -path "./vendor/*")
 DOCKER_IMAGE_NAME ?= thanos
 DOCKER_IMAGE_TAG  ?= $(subst /,-,$(shell git rev-parse --abbrev-ref HEAD))-$(shell date +%Y-%m-%d)-$(shell git rev-parse --short HEAD)
+FIRST_GOPATH      ?= $(firstword $(subst :, ,$(shell go env GOPATH)))
+PROMU             ?= $(FIRST_GOPATH)/bin/promu
+GOIMPORTS         ?= $(FIRST_GOPATH)/bin/goimports
+DEP               ?= $(FIRST_GOPATH)/bin/dep
 
 all: install-tools deps format build
 
-deps:
+deps: vendor
+
+vendor: Gopkg.toml Gopkg.lock | $(DEP)
 	@echo ">> dep ensure"
-	@dep ensure
+	@$(DEP) ensure
 
-format:
+format: $(GOIMPORTS)
 	@echo ">> formatting code"
-	@goimports -w $(FILES)
+	@$(GOIMPORTS) -w $(FILES)
 
 vet:
 	@echo ">> vetting code"
 	@go vet ./...
 
-build:
+build: deps $(PROMU)
 	@echo ">> building binaries"
-	@promu build --prefix $(PREFIX)
+	@$(PROMU) build --prefix $(PREFIX)
+
+install-tools: $(PROMU) $(GOIMPORTS) $(DEP)
```

Comment:
we can remove it now, right?

Reply:
I left it because of 2 reasons:
- I didn't want to change the published interface of Makefile, so that people working on Thanos wouldn't be surprised targets were missing;
- I believe the CI pipeline is using this (and other) targets, so it would break it.

Reply:
Sure, but:
> I didn't want to change the published interface of Makefile, so that people working on Thanos wouldn't be surprised targets were missing;

Having two ways of doing one thing is even worse. Plus, we don't have yet 1.0.0 release so in theory API can change (not even mentioning some Makefile command that is most likely used just single time) (:

> I believe the CI pipeline is using this (and other) targets, so it would break it.

We should modify CI pipeline to use new commands (since they should work on CI too - CI need to test them)

Reply:
The only use case I can see is when someone wants to install all tools in one-go, but not sure if useful.

Reply:
Ok, I'll amend the CI config and remove this target.